src/lib.rs

```diff
@@ -731,16 +750,35 @@ impl<T: AsSocket> Async<T> {
     /// # std::io::Result::Ok(()) });
     /// ```
     pub fn new(io: T) -> io::Result<Async<T>> {
-        let borrowed = io.as_socket();
-
         // Put the socket in non-blocking mode.
-        set_nonblocking(borrowed)?;
+        set_nonblocking(io.as_socket())?;
+
+        Self::new_nonblocking(io)
+    }
 
+    /// Creates an async I/O handle without setting it to non-blocking mode.
+    ///
+    /// This method will register the handle in [epoll]/[kqueue]/[event ports]/[IOCP].
+    ///
+    /// On Unix systems, the handle must implement `AsRawFd`, while on Windows it must implement
+    /// `AsRawSocket`.
```

Comment:
Nit: This is supposed to be `AsFd` and `AsSocket`.

Reply:
Good catch. Fixed.